### ISSUE TITLE
Build failure on SmartOS and FreeBSD (6.1)

### ISSUE THREAD

**Issue by targos** [nodejs/node-v8#14]
CI runs:
https://ci.nodejs.org/job/node-test-commit-smartos/10603/nodes=smartos15-64/console
https://ci.nodejs.org/job/node-test-commit-freebsd/10730/nodes=freebsd11-x64/console

SmartOS error:

```
../deps/v8/src/base/platform/platform-solaris.cc:61:7: error: prototype for 'void* v8::base::OS::Allocate(std::size_t, std::size_t*, v8::base::OS::MemoryPermission)' does not match any in class 'v8::base::OS'
 void* OS::Allocate(const size_t requested, size_t* allocated,
       ^
In file included from ../deps/v8/src/base/platform/platform-posix.h:8:0,
                 from ../deps/v8/src/base/platform/platform-solaris.cc:31:
../deps/v8/src/base/platform/platform.h:172:16: error: candidates are: static void* v8::base::OS::Allocate(std::size_t, std::size_t*, bool, void*)
   static void* Allocate(const size_t requested, size_t* allocated,
                ^
../deps/v8/src/base/platform/platform.h:167:16: error:                 static void* v8::base::OS::Allocate(std::size_t, std::size_t*, v8::base::OS::MemoryPermission, void*)
   static void* Allocate(const size_t requested, size_t* allocated,
                ^
deps/v8/src/v8_libbase.target.mk:109: recipe for target '/home/iojs/build/workspace/node-test-commit-smartos/nodes/smartos15-64/out/Release/obj.target/v8_libbase/deps/v8/src/base/platform/platform-solaris.o' failed
```

FreeBSD error:

```
../deps/v8/src/base/platform/platform-freebsd.cc:43:11: error: out-of-line definition of 'Allocate' does not match any declaration in 'v8::base::OS'
void* OS::Allocate(const size_t requested, size_t* allocated,
          ^~~~~~~~
1 error generated.
deps/v8/src/v8_libbase.target.mk:120: recipe for target '/usr/home/iojs/build/workspace/node-test-commit-freebsd/nodes/freebsd11-x64/out/Release/obj.target/v8_libbase/deps/v8/src/base/platform/platform-freebsd.o' failed
```

/cc @nodejs/platform-freebsd @nodejs/platform-smartos

**Comment by targos** [comment]
V8 6.1 is also affected.

**Comment by targos** [comment]
I think I have the fix

**Comment by targos** [comment]
CL: https://chromium-review.googlesource.com/c/600194

**Comment by targos** [comment]
CL landed and is going to be merged to 6.1.
